### PR TITLE
Create WorkspaceViewport class

### PR DESCRIPTION
**Pull request by rachel-fenichel**
This class has helper functions for figuring out the size of the viewport into the workspace, and is used by WorkspaceSvg.

This is where code will go to emit events when the workspace viewport changes.  It may also be responsible for some of the workspace scrolling code.

### CODE REVIEW

File: core/workspace_viewport.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    width: width,
+    height: height
+  };
+};
+
+/**
+ * Get the content dimensions of the given workspace, taking into account
+ * whether or not it is scrollable and what size the workspace div is on screen.
+ * @param {!Blockly.WorkspaceSvg} ws The workspace to measure.
+ * @param {!Object} svgSize An object containing height and width attributes in
```
Comment (RoboErikG) [inline core/workspace_viewport.js]: height and width attributes of what? the workspace div?

File: core/workspace_viewport.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    width: width,
+    height: height
+  };
+};
+
+/**
+ * Calculate the size of a scrollable workspace, which should include room for a
+ * half screen border around the workspace contents.
+ * @param {!Blockly.WorkspaceSvg} ws The workspace to measure.
+ * @param {!Object} svgSize An object containing height and width attributes in
```
Comment (RoboErikG) [inline core/workspace_viewport.js]: Same question as above.

**Comment by rachel-fenichel** [comment]
Updated, PTAL.

**Comment by picklesrus** [comment]
I'd like to get a better understanding of your future plans for this class. This one doesn't have any state and is just a collection of utilities right now.  I'm all for breaking up the code into smaller pieces, but I can't see the bigger picture on this one yet and in my current understanding getDimensionsPx seems misplaced.  I just want to be careful before adding more publicly usable functions.

**Comment by rachel-fenichel** [comment]
I talked to @picklesrus about this and I'm going to move the functions back into workspace_svg.js for now so that I'm not adding a new public interface.  They'll still be decomposed, and I may create the WorkspaceViewport class in the future if I still see a need for it.

**Comment by rachel-fenichel** [comment]
Updated.  I'll push when travis is happy.

**Pull request merged**